src/gsi/general_read_gfsatm.f90

```diff
@@ -2073,57 +2161,73 @@ subroutine general_read_gfsatm_nc(grd,sp_a,filename,uvflag,vordivflag,zflag, &
-   if ( istatus /= 0 ) then
-      if ( mype == 0 ) then
-         write(6,*) 'general_read_gfsatm_nems: ERROR'
-         write(6,*) 'Missing some of the required fields'
-         write(6,*) 'Aborting ... '
-      endif
-      call stop2(999)
+   if (.not. read_2m) then
+       iredundant=0
+       call gsi_bundlegetpointer(gfs_bundle,'sf',g_div ,ier)
+       if ( ier == 0 ) iredundant = iredundant + 1
+       call gsi_bundlegetpointer(gfs_bundle,'div',g_div ,ier)
+       if ( ier == 0 ) iredundant = iredundant + 1
+       if ( iredundant==2 ) then
+          if ( mype == 0 ) then
+             write(6,*) 'general_read_gfsatm_nems: ERROR'
+             write(6,*) 'cannot handle having both sf and div'
+             write(6,*) 'Aborting ... '
+          endif
+          call stop2(999)
+       endif
+       iredundant=0
+       call gsi_bundlegetpointer(gfs_bundle,'vp',g_vor ,ier)
+       if ( ier == 0 ) iredundant = iredundant + 1
+       call gsi_bundlegetpointer(gfs_bundle,'vor',g_vor ,ier)
+       if ( ier == 0 ) iredundant = iredundant + 1
+       if ( iredundant==2 ) then
+          if ( mype == 0 ) then
+             write(6,*) 'general_read_gfsatm_nems: ERROR'
+             write(6,*) 'cannot handle having both vp and vor'
+             write(6,*) 'Aborting ... '
+          endif
+          call stop2(999)
+       endif
+       iredundant=0
+       call gsi_bundlegetpointer(gfs_bundle,'t' ,g_tv  ,ier)
+       if ( ier == 0 ) iredundant = iredundant + 1
+       call gsi_bundlegetpointer(gfs_bundle,'tv',g_tv  ,ier)
+       if ( ier == 0 ) iredundant = iredundant + 1
+       if ( iredundant==2 ) then
+          if ( mype == 0 ) then
+             write(6,*) 'general_read_gfsatm_nems: ERROR'
+             write(6,*) 'cannot handle having both t and tv'
+             write(6,*) 'Aborting ... '
+          endif
+          call stop2(999)
+       endif
+
+       istatus=0
+       call gsi_bundlegetpointer(gfs_bundle,'ps',g_ps  ,ier);istatus=istatus+ier
+       call gsi_bundlegetpointer(gfs_bundle,'q' ,g_q   ,ier);istatus=istatus+ier
+       call gsi_bundlegetpointer(gfs_bundle,'oz',g_oz  ,ier);istatus=istatus+ier
+       call gsi_bundlegetpointer(gfs_bundle,'cw',g_cwmr,ier);istatus=istatus+ier
+       if ( istatus /= 0 ) then
+          if ( mype == 0 ) then
+             write(6,*) 'general_read_gfsatm_nems: ERROR'
+             write(6,*) 'Missing some of the required fields'
+             write(6,*) 'Aborting ... '
+          endif
+          call stop2(999)
+       endif
+   else ! read 2m vars
+       istatus=0
+       call gsi_bundlegetpointer(gfs_bundle,'t2m',g_t2m  ,ier);istatus=istatus+ier
+       call gsi_bundlegetpointer(gfs_bundle,'q2m',g_q2m  ,ier);istatus=istatus+ier
+       call gsi_bundlegetpointer(gfs_bundle,'ps',g_ps  ,ier);istatus=istatus+ier
+       if ( istatus /= 0 ) then
+          if ( mype == 0 ) then
+             write(6,*) 'general_read_gfsatm_nems: ERROR'
```

Comment:
Minor: It looks like you inherited this from the previous code, but the subroutine is general*_nc, not general*_nems.